### ISSUE TITLE
[Lecture5][0405] Understanding the array appended to A in Homography's function

### ISSUE THREAD

**Issue by Burugi** [PiLab-CAU/ComputerVision-2401#10]
![image](https://github.com/PiLab-CAU/ComputerVision-2401/assets/52272632/fb7a88d3-122a-4423-a558-73014e67fea9)

When I was trying to find the homography in the picture above, I was told that the array that A.appends to would make sense if I tried it.
I created an issue to confirm my understanding and to share it with other students.
Here's what I want to write

1. the calculation for A.append([-x.-y,-1,0,0,0,0,x*u,y*v,u])
2. why only 4 points are needed as a minimum

Also, as I understand it, I would like to know why it is okay to leave h_22 at 1.
![image](https://github.com/PiLab-CAU/ComputerVision-2401/assets/52272632/4964780f-a6f7-4a98-8f32-087d0108c1f8)

1. probably a picture will explain the formula, but I'll use English again.
Light blue writing -> A matrix of A with n dots appended, and h is the value of zero.
If we take the first row of A as an example, it looks like this
What we need to be satisfied with is that H*(src point) = (dst point), and if we arrange the expressions, we get the three expressions on the far right of the picture.
You can see that the first row has the expression for the condition in the calculation.
So if we calculate it, we can see in red that -u_1+1*u_1=0.

If we extend this to v_1, we can see that the equation is satisfied.

2. So why are there at least 4 possible points?
There are a total of 9 unknowns in H that we need to know. But since you said it's okay to let h_22 =1 here, we have a total of 8 unknowns.
For every point, we have 2 given points, so we need at least 4 points.

However, I still don't understand why it's okay to have h_22 = 1.

I would like you to confirm if the following understanding and explanation is correct.

Thank you.

**Comment by yjyoo3312** [comment]
@Burugi 
Thanks for your feedback! Due to time constraints, I had to bypass it earlier. I've added some extra details below, but in essence, here's what we're aiming for:

- Our goal is to shape our equation into the form Ah = 0, allowing us to apply SVD (Singular Value Decomposition).
- To achieve this, we treat h_22 (which is initially set to 1) as a variable, denoting it as k, within the framework of homogeneous coordinates.
- In the end, we arrive at an equation Ah'=0, where h' includes 8 unknowns plus one additional parameter (h_22). We then obtain the final homography by normalizing with respect to the augmented h_22.

<img width="1191" alt="image" src="https://github.com/PiLab-CAU/ComputerVision-2401/assets/12525981/af2001f7-5b0b-43ff-bb1d-ed712ed8dd0d">

Hope this explanation helps! :)
